t/t3907-stash-show-config.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/sh
+
+test_description='Test git stash show configuration.
+
+'
+
+. ./test-lib.sh
+
+config_stash_stat_and_patch () {
+	if test "$1" != "unset"
+	then
+		git config stash.showStat "$1"
+	fi
+
+	if test "$2" != "unset"
+	then
+		git config stash.showPatch "$2"
+	fi
+}
+
+unset_config () {
+	if git config --bool stash.showStat >/dev/null
+	then
+		git config --unset stash.showStat
+	fi
+
+	if git config --bool stash.showPatch >/dev/null
+	then
+		git config --unset stash.showPatch
+	fi
+}
+
+get_expected_output () {
+	FLAGS=
+
+	if test "$(git config --bool stash.showStat || echo true)" = "true"
```

Comment:
I *think* we have some new option for `git config` that defines a default value...

*clicketyclick*

Yep: https://git-scm.com/docs/git-config#git-config---defaultltvaluegt

So: `test true = "$(git config --bool --default true stash.showStat)"` will get you where you want to be.